### DISCUSSION TITLE
Windowed Comparator across multiple pins?

### DISCUSSION THREAD

**Discussion by Gabe-From-Onda** [SpenceKonde/megaTinyCore#632] [Q&A]
I'm programming an Attiny1626 to have it monitor multiple pins to see if they go outside of a voltage range and send a serial message with the pin number and whether it went above or below the threshold.

The Windowed Comparator looks like it would be a super useful thing, but it looks like there's only 1, and I'd have to loop the MUX between each pin to have the comparator look at each voltage. Having the additional code to handle all this doesn't seem like it's that much better than something along the lines of:

for(int i=0; i<7; i++){
     val = analogRead(pin[i]);
     if(val < lowVal){
          Serial.print(low[i]);
     }
     else if(val > highVal){
          Serial.print(high[i]);
     }
}

I don't see a huge benefit to using the Windowed Comparator here. However, I'm often wrong and would love to be surprised with a better way to handle this.

**Comment by SpenceKonde** [comment]
I agree with your assessment. It is only useful when you need to monitor only a single pin. I'll add that it's also only useful if you don't need the ADC for anything else and hence can afford to put it into free running mode, and when it is appropriate in terms of your requirements to have the response to this condition be the firing of an interrupt (ie if you need to respond quickly). 

It's more useful on the tinyAVR 1-series because they have two identical ADCs. Likely the intent was for people who wanted a differential ADC which the 1-series doesn't have could have them both trigger conversions at the same moment, and subtract the results from eachother - but it also means you can use the first one normally (analogRead only uses ADC0) and have ADC1 in free running mode with the window comparator to trigger an interrupt when the monitored pin has an excursion from it's specified values.
